R/install.R

```diff
@@ -396,3 +398,172 @@ build_status_ok <- function(process_log, quiet = FALSE) {
+        } else {
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
+          write('PATH="${RTOOLS40_HOME}\\usr\\bin;${RTOOLS40_HOME}\\mingw64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          stop("Restart R and run check_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.", call. = FALSE)
+        }
+      }
+    } else {
+      mingw32_make_path <- dirname(Sys.which("mingw32-make"))
+      gpp_path <- dirname(Sys.which("g++"))
+      # If mingw32-make and g++ are not found, we check typical RTools 3.5 folders.
+      # If found, we fix PATH, otherwise we recommend the user to install RTools 3.5.
+      if (!nzchar(mingw32_make_path) || !nzchar(gpp_path)) {
+        rtools_path <- Sys.getenv("RTOOLS35_HOME")
+        if (!nzchar(rtools_path)) {
+          typical_paths <- c(
+            file.path("C:/", "Rtools"),
+            file.path("C:/", "Rtools35"),
+            file.path("C:/", "RTools"),
+            file.path("C:/", "RTools35"),
+            file.path("C:/", "rtools"),
+            file.path("C:/", "rtools35"),
+            file.path("c:/", "Rtools"),
+            file.path("c:/", "Rtools35"),
+            file.path("c:/", "RTools"),
+            file.path("c:/", "RTools35"),
+            file.path("c:/", "rtools"),
+            file.path("c:/", "rtools35")
+          )
+          for (p in typical_paths) {
+            if (dir.exists(p)) {
+              rtools_path <- p
+            }
+          }
+        }
+        if (nzchar(rtools_path)) {
+          if (!fix) {
+            error_message <- c(
+              "",
+              "A RTools installation was found, but the PATH is not properly set.",
+              "Run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)' to fix the issue."
+            )
+            stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+          }
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
+          if (nzchar(Sys.getenv("RTOOLS35_HOME"))) {
+            write(paste0('RTOOLS35_HOME=', rtools_path), file = "~/.Renviron", append = TRUE)
+          }
+          write('PATH="${RTOOLS35_HOME}\\bin;${RTOOLS35_HOME}\\mingw_64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          stop("Restart R and run check_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.", call. = FALSE)
+        } else {
+          error_message <- c(
+            "",
+            "A toolchain was not found. Please install RTools 3.5 and run",
+            "",
+            "write(\'RTOOLS35_HOME=rtools35/install/path/\', file = \"~/.Renviron\", append = TRUE)",
+            "replacing 'rtools35/install/path/' with the actual install path of RTools 3.5.",
+            "",
+            "Then restart R and run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)'."
+          )
+          stop(paste0(error_message, collapse = "\n"), call. = FALSE)
```

Comment:
Oh, now I think I see why you're using the blank `""` in most of these error messages. If you prefer this way that's fine, but you can get the same output with fewer lines of code like this: 

```
stop(
  "A toolchain was not found. Please install RTools 3.5 and run",
  "\n\nwrite(\'RTOOLS35_HOME=rtools35/install/path/\', file = \"~/.Renviron\", append = TRUE)",
  "\nreplacing 'rtools35/install/path/' with the actual install path of RTools 3.5.",
  "\n\nThen restart R and run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)'.", 
  call. = FALSE
)
```

In either case, `'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)'` can be replaced with 
`check_cmdstan_toolchain(fix = TRUE)`

Reply:
Thanks, replaced with your suggestion.